octavia_f5/controller/statusmanager/status_manager.py

```diff
@@ -253,65 +251,58 @@ def _get_lb_msg(lb_id):
             self.health_executor.submit(update_health, msg)
             self.stats_executor.submit(update_stats, msg)
 
-    @rollback_on_db_exception
-    def update_listener_count(self, num_listeners, session=None):
+    def update_listener_count(self, num_listeners):
         """ updates listener count of bigip device (vrrp_priority column in amphora table)
 
-        :param session: Database session supplied by the rollback_on_db_exception wrapper.
-                        Don't supply this argument when calling this function!
         :param num_listeners: number of listener for the bigip device
         """
-        device_name = self.active_bigip.hostname
-        device_entry = self.amp_repo.get(session,
-                                         compute_flavor=CONF.host,
-                                         load_balancer_id=None,
-                                         cached_zone=device_name)
-        if not device_entry:
-            self.amp_repo.create(
-                session,
-                compute_flavor=CONF.host,
-                vrrp_priority=num_listeners,
-                cached_zone=device_name)
-        else:
-            self.amp_repo.update(session, device_entry.id,
-                                 vrrp_priority=num_listeners)
-        session.commit()
-
-    @rollback_on_db_exception
-    def update_availability(self, device_name, available, session=None):
+        with DatabaseLockSession as session:
```

Comment:
You need to instantiate context manager class DatabaseLockSession, else getting exception

```
2020-04-16 16:53:39.616 11252 ERROR futurist.periodics   File "/Users/d072895/Workspace/stein/octavia-f5-provider-driver/octavia_f5/controller/statusmanager/status_manager.py", line 260, in update_listener_count
2020-04-16 16:53:39.616 11252 ERROR futurist.periodics     with DatabaseLockSession as session:
2020-04-16 16:53:39.616 11252 ERROR futurist.periodics AttributeError: __exit__
```